### PR TITLE
CI: update `--use-local-cdk` arg in `airbyte-ci`, adds `--use-cdk-ref` arg

### PR DESCRIPTION
**Pull request by aaronsteers**
## What

This PR updates `--use-local-cdk` behavior for Python connectors and adds `--use-cdk-ref` flag.

Testing here:

- https://github.com/airbytehq/airbyte-python-cdk/pull/40

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by aaronsteers** [comment]
@alafanechere - This is only a start. 
Do you have time to assist?

If `poetry add` doesn't work, we could revert to the `pip inject` method, but just with an alternate path.

File: airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py [hunk 1]
```diff
@@ -242,16 +242,16 @@ async def apply_python_development_overrides(context: ConnectorContext, connecto
     if context.use_local_cdk:
         context.logger.info("Using local CDK")
         # mount the local cdk
-        path_to_cdk = "airbyte-cdk/python/"
+        path_to_cdk = "../airbyte-python-cdk"  # Assume CDK is cloned in a sibling dir to `airbyte`
         directory_to_mount = context.get_repo_dir(path_to_cdk)
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py:247]: `get_repo_dir` relates to the `airbyte` repo.
To get a directory on the host in the native dagger way you can do like this:

```suggestion
        directory_to_mount = context.dagger_client.host().directory(path_to_cdk)
```

(the relative path might not work, but an absolute path will for sure)

File: airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py [hunk 2]
```diff
@@ -242,16 +242,16 @@ async def apply_python_development_overrides(context: ConnectorContext, connecto
...
 
         context.logger.info(f"Mounting CDK from {directory_to_mount}")
 
         # Install the airbyte-cdk package from the local directory
-        # We use `--force-reinstall` to use local CDK with the latest updates and dependencies
-        connector_container = connector_container.with_mounted_directory(f"/{path_to_cdk}", directory_to_mount).with_exec(
-            ["pip", "install", "--force-reinstall", f"/{path_to_cdk}"]
-        )
+        connector_container = connector_container.with_mounted_directory(
+            path_to_cdk,
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py]:
```suggestion
            "/cdk",
```

File: airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py [hunk 3]
```diff
@@ -242,16 +242,16 @@ async def apply_python_development_overrides(context: ConnectorContext, connecto
...
+            directory_to_mount,
+        ).with_exec(["poetry", "add", f"--path={path_to_cdk}"])
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py]:
```suggestion
        connector_container = connector_container.with_mounted_directory(
            path_to_cdk,
            directory_to_mount,
        ).with_exec(["poetry", "add", "--path=/cdk"])
```

Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py]: nit:
If you don't want to make the success of the execution depend on a pre-cloned CDK repo you can do somehting like this:

(only works on the CDK repo and on GHA)

```python
cdk_ref = os.environ["GITHUB_REF"] 
...
.with_exec(["poetry", "add", f"git+https://github.com/airbytehq/airbyte-python-cdk.git#{cdk_ref}"])
```

**Review by alafanechere — CHANGES_REQUESTED** [review]
Requesting changes as your directory mounting logic will not work as it is. Feel free to force merge once you fixed it. 
And please bump the `pipelines`  package version + add an entry to the changelog in README.md

File: airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py (hunk 3 above)
Comment (aaronsteers) [reply airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py]: Now that cdk local is working well, I added a new arg `--use-cdk-ref` that allows branches and tags to be specified.

Comment (aaronsteers) [reply airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py]: Updated to be more explicit about which paths are used for which.

File: airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py [hunk 4]
```diff
@@ -242,16 +242,16 @@ async def apply_python_development_overrides(context: ConnectorContext, connecto
     if context.use_local_cdk:
         context.logger.info("Using local CDK")
         # mount the local cdk
-        path_to_cdk = "airbyte-cdk/python/"
+        path_to_cdk = "../airbyte-python-cdk"  # Assume CDK is cloned in a sibling dir to `airbyte`
```
[the remark this answers is not in the document]
Comment (aaronsteers) [reply airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py]: Done 👍

**Comment by aaronsteers** [comment]
@alafanechere - Tests are looking good so I'm yolo-merging this to try to resolve a couple other issues. Will gather data as we run in a variety of environments, and will try to resolve what comes up.

**Comment by aaronsteers** [comment]
Intentionally un-deleting branch (for now) because we have a CDK pipeline pinned to it.

**Pull request merged**